### ISSUE TITLE
`jfrog rt docker-push` does not work when using `--apikey`

### ISSUE THREAD

**Issue by djgilcrease** [jfrog/jfrog-cli-go#352]
```bash
$ jfrog rt docker-push --url https://artifactory.example.com/artifactory --apikey stuff --build-name docker-img --build-number 1.2.0 artifactory.example.com/cicd-docker/docker-img:d2e40a5ed58eae04750c6ecc2f3f6f0019a7981d cicd-docker
Must provide --username with --password-stdin
Must provide --username with --password-stdin
[Error] Docker login failed for: artifactory.example.com/cicd-docker, artifactory.example.com.
$ docker push artifactory.example.com/cicd-docker/docker-img:d2e40a5ed58eae04750c6ecc2f3f6f0019a7981d
The push refers to repository [artifactory.example.com/cicd-docker/docker-img]
866f0fff9cea: Layer already exists 
70f3555fd979: Layer already exists 
085b28397dee: Layer already exists 
cc163dcea492: Layer already exists 
82c4945bae82: Layer already exists 
a55dcc055071: Layer already exists 
6284ba0dafcb: Layer already exists 
4fd277466317: Layer already exists 
bcf2f368fe23: Layer already exists 
d2e40a5ed58eae04750c6ecc2f3f6f0019a7981d: digest: sha256:32958c0414efe8e376b5313e2c018074a501ae1c1f3ebbce6b2c436559aaf2a3 size: 2197
```

**Comment by eyalbe4** [comment]
@djgilcrease,
Does adding the ```--user``` option to the command resolve the issue?
The docker client expects a username and a password (or an API key). We also recommend using API key only with a username this is the common convention.
In addition, the ```jfrog rt c``` command no longer accepts an API key without a username.

**Comment by djgilcrease** [comment]
No, even when passing --user and --apikey it does not work. It requires --user && --password

{code}
$ jfrog --version
jfrog version 1.24.2
{code}

**Comment by djgilcrease** [comment]
resolved by #380
